### ISSUE TITLE
"login" brings user to the https://gui.dandiarchive.org/#/

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandiarchive#679]
If I am already in a dandiset (e.g. https://gui.dandiarchive.org/#/dandiset/000029), and click logout -- all good, I stay in the same page.

If I try to re-login again then two things happen and possibly related thus filing together
- I get forwarded to "Authorize DANDI GUI?" https://api.dandiarchive.org/oauth/authorize/?redirect_uri=https%3A%2F%2Fgui.dandiarchive.org%2F&client_id=...  URL (so `redirect_uri` does not carry dandiset any longer), even though I did authorize github before (so I expect that to not happen again or for a while)
- and when I click authorize I get to the top landing page

**Comment by waxlamp** [comment]
I think we can come up with a simple way to accomplish this.

**Comment by sanuann** [comment]
right now, when trying to re-login I do not see the "Authorize DANDI GUI" page. it just logs me in but brings me to the top landing page. so I am assuming this is not yet fixed?
Can I work on this @waxlamp?

**Comment by yarikoptic** [comment]
@waxlamp IIRC already addressed having not-needed "Authorize DANDI GUI" dialog. The original issue (not going to the original page) indeed was not resolved yet AFAIK.

**Comment by waxlamp** [comment]
Right, the problem now is simply that if you're logged out on a DLP, and then you log in, you are redirected to the front page rather than where you were.

I can think of some hacky ways to fix this that include saving the URL in local storage when the login button is clicked, but I don't know if that's a bad idea. If you want to work on this, please consult with @dchiquito about solution strategies before you try to solve this.

**Comment by yarikoptic** [comment]
I bet there is some option to that github authorization endpoint to provide a redirect url. That redirect url should just be the original url where "LOGIN VIA GITHUB" is clicked. ... yeap: https://docs.github.com/en/developers/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps#1-request-a-users-github-identity 

redirect_uri | string | The URL in your application where users will be sent after authorization. This must be an exact match to one of the URLs you provided as a Callback URL when setting up your GitHub App and can't contain any additional parameters.
-- | -- | --

isn't that the one?

**Comment by waxlamp** [comment]
> edit: ha ha -- read it in full and **exact match** I guess ruins it all... so it would require some state (yet another cookie?) remembered in web ui and then recalled/removed whenever that callback end point is hit, and which redirects to that target page

The naive thing I can think of is to record `window.location` in `localStorage` at the time the login button is clicked, go through the auth workflow, and then when the user arrives back in the app, check for the recorded URL and navigate there immediately if found.

This is likely somewhat fragile. But it would work, and of course if there are better ways we should use them.

**Comment by dchiquito** [comment]
I think localStorage is probably the only way to do this.
